### PR TITLE
vfat.io - Sickle

### PR DESCRIPTION
**Pull request by 0xChupaCabra**
This pr is just an update for the pancakeswap adapter.We deployed our subgraph at https://graph.vf.at/subgraphs/name/sickle/graphql . It returns a list of all deployed Sickle contracts. Then we update the final result with the Sickle owner address.

### CODE REVIEW

File: adapters/pancakeswap/src/index.ts [hunk 1]
```diff
@@ -78,7 +80,27 @@ export const getUserTVLByBlock = async ({
...
   }
 
+  // Get sickles and their owners
+  const sickleAddresses = await getSickles();
+  const sickleOwners = await getSickleOwners(sickleAddresses.map(s => s.sickle));
+
+  // Replace sickle addresses with their owners
+  const updatedBalances: Record<string, Record<string, bigint>> = {};
   for (const [user, tokenBalances] of Object.entries(balances)) {
+    const owner = (sickleOwners as Record<string, string>)[user] || user; // Replace sickle address with owner address
```
Comment (chef-jojo) [inline adapters/pancakeswap/src/index.ts:90]: sickleOwners key address is checksummed right? better convert it to lowercase, as balances[user] is lowercase by default in subgraph

File: adapters/pancakeswap/src/sdk/lib.ts [hunk 2]
```diff
@@ -164,6 +165,24 @@ export const getV3UserPositionsAtBlock = async (
   return [...resultMap.values()];
 };
 
+export const getSickles = async () => {
+  const query = `query fetchSickles {
```
Comment (chef-jojo) [inline adapters/pancakeswap/src/sdk/lib.ts:169]: you may need to also provide block number in subgraph
`block: { number: ${blockNumber} },`

Comment (0xChupaCabra) [reply adapters/pancakeswap/src/sdk/lib.ts:169]: Let's say we pass a block number, and the sickle contract Deployed event is before this block, but it has a balance. We will miss the owner in this case. This is why i decided to kerp the function without a block number. Correct me if i am wrong

**Review by chef-jojo — APPROVED** [review]

**Comment by 0xChupaCabra** [comment]
hello @chef-jojo we have also added vfat.io code on the nile adapter. my mistake i made both changes in the same pr

**Comment by nitish-91** [comment]
@0xChupaCabra please use the PR template to fill all the required details. Also can you please explain why you are changing code for 2 protocols, Nile and pancakesawp

**Comment by vfat0** [comment]
> @0xChupaCabra please use the PR template to fill all the required details. Also can you please explain why you are changing code for 2 protocols, Nile and pancakesawp

Hello, our project [vfat.io](vfat.io/yield) is a yield aggregator with 1-tx deposits into other protocols. On Linea we support Pancakeswap and Nile so far. We deploy a proxy contract for each user to do the batching, so since the deposits are held by the proxy they wouldn't be earning XP. 

The PR updates both of these projects to check if the user is one of our contracts (called Sickles) and replaces the Sickle address with its EOA owner. We have discussed this with folks at Linea/Pancakeswap/Nile.

**Comment by 0xroll** [comment]
hey @vfat0 thanks for the PR.

just wondering if its better for you to have them returned under /vfat 's tvl rather than changing the adapters for the protocols.

we can discuss further , please dm me at @jh_onchain

**Comment by 0xroll** [comment]
@vfat0 as discussed, please create another pr for vfat protocol instead of modifying the AMMs, thanks!

**Pull request closed without merging**